pkg/data/source.go

```diff
@@ -0,0 +1,32 @@
+package data
+
+import "fmt"
+
+//DataSource indentifies a format of incoming data in the message bus channel.
+type DataSource int
+
+//ListAll returns slice of supported data sources in human readable names.
+func (src DataSource) ListAll() []string {
```

Comment:
I have not yet explored where you use this type, however I don't really like the idea of adding message type specific logic to the core code base. This data package was really only intended to define types that are used internally by the bus and main process and provide the interfaces so that plugins could use them. In general, nothing within the pkg directory should mention any specific message source like collectd or ceilometer, any transport protocols, or any specific backend applications. Rather, the sg-core only deals with its own generic types while the plugins deal the specific ones. I fear if we start doing this, we will end up with the very uncoupled, tangled code that was in the original SG. Can this be moved to one of the plugin packages?

Reply:
The DataSource is used in alertmanager app and elasticsearch app. If we would want to have that out of sg-core/pkg we have to either have two copies of the definition in each app or create some shared module in sg-core/plugins(maybe?). The first option would make sense if each plugin would have separate repo, but plugins are part of core repo, so copying code over plugins would mean we are breaking DRY principle. The second option is just moving the module to more non-sense (IMO) place. Any other idea where to move it? Maybe even move this to infrawatch/apputils library? Both plugins are importing the library and the shared code would be decoupled from sg-core completely. Does this make sense to you?

Reply:
Yeah that makes sense. The plugins are included in the same project so that building is easier, but they are still basically standalone projects from the sg-core. Granted, I know that ties their version to the sg-core version, but I figured it would be best to keep our main plugins in the same repository and if anyone creates a third party plugin, it can be in their own repository. I'll dig into it more in depth this afternoon and see if I can come up with some alternate solutions. It seems to me that putting addresses in the bus may actually solve our problem here since this object appears to be used mostly to determine the event source type

Reply:
Upon further review, the reason we are running into this problem is because here, logic that should be a part of the event handler is implemented in the application interface. That is really the fault of the internal event interface since it is too generic to work with every application plugin. This interface needs to be refined such that it is generic enough that all important event information can be placed into it by any handler, but restrictive enough that application plugins can always expect the same structure regardless of the source of the event. This is why the metric interface is defined as it is.

Reply:
Based on collectd and ceilometer events, this may work as an event interface:
```
severity string
time (epoch) uint64
message string
metadata map[string]interface{}
```